### PR TITLE
Add server configurations guidelines to WSO2 IS documentation

### PR DESCRIPTION
**Pull request by osandamaleesha**
### **Proposed changes in this pull request**

This PR adds guidelines to wso2 IS documentation for server configurations of action execution addressed in https://github.com/wso2/product-is/issues/21119

Refer to Proposed UI implementation attached here.

Screenshot (38)

Screenshot (39)

### **When should this PR be merged**

Soon

### **Follow up actions**

N/A

### CODE REVIEW

File: en/includes/guides/customize/actions/setting-up-actions.md [hunk 1]
```diff
@@ -70,3 +70,52 @@ The following are some of the troubleshooting steps that you may take to resolve
 1. Ensure that your external service is up and running and that there are no connectivity issues.
 
 2. Confirm that the request and response payloads conform to the expected formats as defined by the REST API contract.
+
+## Configuring HTTP Client Connections
```
Comment (malithie) [inline en/includes/guides/customize/actions/setting-up-actions.md:74]: Shall we say 'Fine tune HTTP Client Connections'

File: en/includes/guides/customize/actions/understanding-actions.md [hunk 2]
```diff
@@ -175,6 +173,9 @@ If the error response includes an acceptable payload, it is treated as an error
 </tbody>
 </table>
 
+!!! note
+    Refer to [Configuring HTTP Client Connections]({{base_path}}/guides/customize/actions/setting-up-actions/#configuring-http-client-connections) to adjust timeouts, connection pooling, and retries in deployment.toml for optimal performance.
```
Comment (malithie) [inline en/includes/guides/customize/actions/understanding-actions.md:177]: Let's update the reference here accordingly

File: en/includes/guides/customize/actions/setting-up-actions.md (hunk 1 above)
Comment (osandamaleesha) [reply en/includes/guides/customize/actions/setting-up-actions.md:74]: Updated in

File: en/includes/guides/customize/actions/understanding-actions.md (hunk 2 above)
Comment (osandamaleesha) [reply en/includes/guides/customize/actions/understanding-actions.md:177]: Updated in

**Comment by malithie** [comment]
@osandamaleesha can you update the screen shots above to reflect latest changes

**Review by malithie — APPROVED** [review]
Overall content address the scope of the improvement done. Open for information architecture and content review further

**Review by himeshsiriwardana — APPROVED** [review]

**Pull request merged**